scripts/metric_reporter/parser/coverage_json_parser.py

```diff
@@ -162,20 +170,21 @@ def parse(self, artifact_file_paths: list[Path]) -> list[CoverageJsonGroup]:
                          JSON data.
         """
         coverage_json_groups: list[CoverageJsonGroup] = []
-        for artifact_file_path in artifact_file_paths:
-            self.logger.info(f"Parsing {artifact_file_path}")
-            file: ArtifactFile = self._parse_artifact_file_name(artifact_file_path)
+        for artifact_file_name in artifact_file_names:
+            self.logger.info(f"Parsing {artifact_file_name}")
+            file: ArtifactFile = self._parse_artifact_file_name(artifact_file_name)
             group: CoverageJsonGroup = self._get_coverage_json_group(file, coverage_json_groups)
             try:
-                with file.path.open() as json_file:
-                    json_data: dict[str, Any] = json.load(json_file)
-                    coverage_json: CoverageJson = self._parse_json_data(file, json_data)
-                    group.coverage_jsons.append(coverage_json)
-            except (OSError, JSONDecodeError, ValidationError) as error:
+                content: str = self._gcs_client.get_coverage_artifact_content(
+                    file.repository, artifact_file_name
+                )
+                json_data: dict[str, Any] = json.loads(content)
+                coverage_json: CoverageJson = self._parse_json_data(file, json_data)
+                group.coverage_jsons.append(coverage_json)
+            except (JSONDecodeError, ValidationError) as error:
```

Comment:
Are you not expecting any other error types here even with GCS?

Reply:
There could be some yes, but I couldn't get them to generate reasonably. So figured to let the higher level exception clause catch them for now.